beater/middleware/auth_middleware.go

```diff
@@ -82,6 +83,27 @@ func AuthMiddleware(authenticator Authenticator, required bool) Middleware {
 	}
 }
 
+// FirehoseAuthMiddleware returns a middleware taking care of authorization header for Firehose
+func FirehoseAuthMiddleware() Middleware {
+	return func(h request.Handler) (request.Handler, error) {
+		return func(c *request.Context) {
+			// Create Authorization header using X-Amz-Firehose-Access-Key header
+			// This is only for Firehose HTTP request
+			accessKey := c.Request.Header.Get("X-Amz-Firehose-Access-Key")
+			var err error
+			if accessKey == "" {
+				id := request.IDResponseErrorsUnauthorized
+				c.Logger.Error(request.MapResultIDToStatus[id].Keyword, logp.Error(err))
+				c.Result.SetWithError(id, err)
+				c.Write()
+				return
+			}
+			c.Request.Header.Add("Authorization", "ApiKey "+accessKey)
+			h(c)
+		}, nil
+	}
+}
```

Comment:
```suggestion
```

I think this might be a holdover? As you're using the authenticator directly in the handler (which I think is fine; it's not cross cutting, there's just one firehose endpoint), you don't need the middleware.

Reply:
Yes I just removed `FirehoseAuthMiddleware` function and changed `firehoseMiddleware`. Thanks for pointing it out!!!